### PR TITLE
[for 5.9.1 only] Revert "Set VBO hints in more places"

### LINKED ISSUE #14999: minetest 5.9 mapblock flickering depending on camera view
### Minetest version

```true
Minetest 5.9.0 (Windows)
Using LuaJIT 2.1.1706185428
Built by Clang 17.0
Running on Windows/10.0.19041 x86_64
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="."
STATIC_LOCALEDIR="locale"
```

### Irrlicht device

win32

### Operating system and version

win10

### CPU model

_No response_

### GPU model

nvidia geforce gtx 970

### Active renderer

OpenGL 4.6.0

### Summary

https://github.com/user-attachments/assets/73dfd5bc-e4ac-4176-8b80-e3dfa5785100

https://github.com/user-attachments/assets/2048482e-a9f7-4e2c-b695-d0f31e3a35e7

### Steps to reproduce

methods:  

1. join creativeworks server with a 5.9 client (maybe on windows only? havent checked other platforms yet) and fly along the rail system

2. [creativeworks_snippet_world.zip](https://github.com/user-attachments/files/16645941/creativeworks_snippet_world.zip) use this world snippet (made via local map saving), make sure you have farlands reloaded game and advtrains on the world, then teleport to `373, 8, -31` to replicate the first video

### PR DESCRIPTION
**Pull request by grorp**
Fixes #14999 for 5.9.1 as discussed on IRC (https://irc.minetest.net/minetest-dev/2024-09-10#i_6200329)

This revert is equivalent to what @wsor4035 did in https://github.com/minetest/minetest/issues/14999#issuecomment-2336503895

## To do

This PR is a Ready for Review.

## How to test

Verify that the issue is gone if you can reproduce it (I can't)

@wsor4035 @nininik0 @TechnoWolfTV feel free to test

### CODE REVIEW

**Comment by sfan5** [comment]
Reverting is okay for 5.9.1 but this not a real fix, we shouldn't merge it to master.
At the very least we should find out which of the five function calls causes the problem.
Perhaps #15144 already somehow solves.

**Comment by grorp** [comment]
Sure, someone can just cherry-pick this into the backport branch and close this PR afterwards.

File: src/itemdef.cpp [hunk 1]
```diff
@@ -474,6 +474,7 @@ class CItemDefManager: public IWritableItemDefManager
 		// Create new ClientCached
 		auto cc = std::make_unique<ClientCached>();
 
+		// Create an inventory texture
```
Comment (sfan5) [inline src/itemdef.cpp:477]:
```suggestion
```

**Comment by wsor4035** [comment]
tested works, will retest once in 5.9.1 pr
will test sfan5 [other pr](https://github.com/minetest/minetest/pull/15144) once its review is resolved

**Comment by SmallJoker** [comment]
Now included in #15137.

**Pull request closed without merging**